src/LodeRunner.API/src/Controllers/ClientsController.cs

```diff
@@ -43,21 +44,20 @@ public ClientsController()
         /// <returns>IActionResult.</returns>
         [HttpGet]
         [SwaggerResponse((int)HttpStatusCode.OK, "Array of `Client` documents or empty array if not found.", typeof(Client[]), "application/json")]
+        [SwaggerResponse((int)HttpStatusCode.NoContent, "`Data not found.`", null, "application/json")]
         [SwaggerResponse((int)HttpStatusCode.ServiceUnavailable, SystemConstants.TerminationDescription)]
         [SwaggerOperation(
             Summary = "Gets a JSON array of Client objects",
             Description = "Returns an array of `Client` documents",
             OperationId = "GetClients")]
-        public IActionResult GetClients([FromServices] ILRAPICache appCache, [FromServices] CancellationTokenSource cancellationTokenSource)
+        public async Task<ActionResult<IEnumerable<Client>>> GetClients([FromServices] ILRAPICache appCache, [FromServices] CancellationTokenSource cancellationTokenSource)
         {
             if (cancellationTokenSource != null && cancellationTokenSource.IsCancellationRequested)
             {
-                return ResultHandler.CreateCancellationInProgressResult();
+                return await ResultHandler.CreateCancellationInProgressResult();
             }
 
-            IEnumerable<Client> clients = appCache.GetClients();
-
-            return appCache.HandleCacheResult(clients, Logger);
+            return await appCache.GetClients(Logger);
```

Comment:
I thought we were going to use the local logger for the cache object?

Reply:
Fixed